src/Dfc.CourseDirectory.Web/Controllers/AddCourseController.cs

```diff
@@ -883,12 +880,14 @@ public async Task<IActionResult> AcceptAndPublish()
 
             if (result.IsSuccess && result.HasValue)
             {
-                return RedirectToAction("Index", "ProviderCourses",
-                    new {
-                        CourseRunId = courseRuns[0].id,
-                        NotificationTitle = "Course added",
-                        NotificationMessage = "You added",
-                    });
+                Session.SetObject(SessionPublishedCourse, new PublishedCourseViewModel
```

Comment:
Ok, so this is a pretty dirty abuse of the session state to pass the data to the next page, but it means that we can ensure the confirmation is only displayed once if the user clicks refresh then it doesn't attempt to publish the course again.